solidity/ecdsa/README.adoc

```diff
@@ -15,7 +17,370 @@ endif::[]
+This transaction locks the sortition pool and sends a request to the Random
+Beacon for a new relay entry. From this moment, no operator can enter
+or leave the pool. Once a new relay entry appears on the chain, all off-chain
+clients perform group selection by calling `WalletRegistry.selectGroup()` view
+function for free. Relay entry provided by the Random Beacon is used as a seed
+for the group selection. After determining signing group members, clients should 
+perform off-chain distributed key generation (DKG).
+<<operator-only,One of the group members>> submits the result to the chain
+calling `WalletRegistry.submitDkgResult(DKG.Result calldata dkgResult)`
+function. Once the result is submitted, a challenge period starts.
+
+During the challenge period, anyone can notify that the submitted DKG result is
+malicious by calling `WalletRegistry.challengeDkgResult(DKG.Result calldata dkgResult)`
+function. A malicious DKG result may contain corrupted data, group members not
+selected by the pool, or incorrect supporting signatures. If such malicious
+result is submitted and successfully challenged, the result submitter gets
+slashed and the malicious result is immediately discarded. The address which
+notified about malicious DKG result is <<punishment,rewarded>>. DKG timeout
+timer is reset, and group members have another chance to submit a valid result.
+
+Once the challenge period passes, and no valid challenge is reported, the DKG
+result submitter should mark the DKG result as approved calling
+`WalletRegistry.approveDkgResult(DKG.Result calldata dkgResult)`.
+This transaction also unlocks the sortition pool.
+The submitter receives an ETH reimbursement for both `submitDkgResult` and
+`approveDkgResult` transaction as described in
+<<transaction-incentives,Transaction Incentives>> section. In case the original
+submitter does not call the `approveDkgResult` function within a specific number
+of blocks, anyone can do that and receive the submitter's reimbursement.
+
+In case the DKG result was not submitted before the timeout, anyone can 
+notify about the timed out DKG by calling `WalletRegistry.notifyDkgTimeout()`
+function and unlock the sortition pool as part of this transaction. 
+In case the relay entry was not produced by the Random Beacon on time,
+anyone can notify a seed timeout by calling `WalletRegistry.notifySeedTimeout()`
+and unlock the sortition pool as a part of this transaction.
+
+Off-chain clients are expected to follow the <<operator-only,submission order>>
+when submitting DKG result to avoid front-running and minimize the cost, but no
+ordering is enforced on-chain.
+
+The sortition pool weights operators by their authorized stake amount and allows
+selecting the same operator multiple times. Inactive/disqualified members during
+the off-chain DKG protocol are marked as ineligible for <<rewards,rewards>> for
+a governable period of time when the DKG result is approved.
```

Comment:
Maybe we coud menrtion `WalletRegistry.notifyOperatorInactivity(Inactivity.Claim calldata claim, uint256 nonce, int32[] calldata groupMembers)` function here?

Reply:
Oh, it's a separate flow in the code. No one needs to call this function in this case. Inactive/disqualified operators are included in the DKG result and are getting marked as ineligible for rewards as a part of `approveDkgResult` transaction.